HalideGenerator.cmake

```diff
@@ -77,10 +77,10 @@ function(halide_add_generator_dependency)
   # generated library
   if (WIN32)
```

Comment:
Is WIN32 a flag for windows, or MSVC specifically? .o still makes sense on windows, if the cmake generator is mingw (or something else).